library/Asana/App.hs

```diff
@@ -65,6 +67,7 @@ loadApp = do
   (appProjectId, appLogLevel, appPerspective, appIgnoreNoCanDo) <-
     execParser $ info (helper <*> optParser) $ fullDesc <> progDesc
       "Report information about an iteration project"
+  let logFunc = error "not initialized"
```

Comment:
I assume you have this because of the chicken-egg forced on you by the `withLogFunc` interface? This was a wart I mentioned in my blog post. You can just directly construct a `LogFunc`, if that'll help avoid this. I can dig up the code in Restyled if you're interested.

Reply:
Yup

Reply:
I did it twice, so take you're pick :)

- https://github.com/restyled-io/restyler/blob/master/src/Restyler/Logger.hs
- https://github.com/restyled-io/restyled.io/blob/master/src/RIO/Logger.hs

The second is more recent so might be better.